feat: add `blas/ext/base/dapxsumkbn-wasm` (#3204)
stdlib-js/stdlib: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md

<!--

@license Apache-2.0

Copyright (c) 2024 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# dapxsumkbn

> Add a scalar constant to each double-precision floating-point strided array element and compute the sum using an improved Kahan–Babuška algorithm.

<section class="usage">

## Usage

```javascript
var dapxsumkbn = require( '@stdlib/blas/ext/base/dapxsumkbn-wasm' );
```

> #### dapxsumkbn.main( N, alpha, x, stride )

Reviewer: ```suggestion
#### dapxsumkbn.main( N, alpha, x, strideX )
```

Author: I suppose the corresponding `blas/base/dasum-wasm` might also need updation as I followed that as a reference for these terminologies.

Reviewer: Yes it does need updating 😬

Adds a scalar constant to each double-precision floating-point strided array element and computes the sum using an improved Kahan–Babuška algorithm.

```javascript
var Float64Array = require( '@stdlib/array/float64' );

var x = new Float64Array( [ 1.0, -2.0, 2.0 ] );

var sum = dapxsumkbn.main( x.length, 5.0, x, 1 );
// returns 16.0
```

The function has the following parameters:

-   **N**: number of indexed elements.
-   **alpha**: scalar constant.
-   **x**: input [`Float64Array`][@stdlib/array/float64].

> -   **strideX**: index increment for `x`.

Reviewer: ```suggestion
-   **strideX**: stride length for `x`.
```

The `N` and stride parameters determine which elements in the strided arrays are accessed at runtime. For example, to access every other element in `x`,

```javascript
var Float64Array = require( '@stdlib/array/float64' );

var x = new Float64Array( [ 1.0, 2.0, 2.0, -7.0, -2.0, 3.0, 4.0, 2.0 ] );

var sum = dapxsumkbn.main( 4, 5.0, x, 2 );
// returns 25.0
```

Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.

<!-- eslint-disable stdlib/capitalized-comments -->

```javascript
var Float64Array = require( '@stdlib/array/float64' );

var x0 = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ] );
var x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element

var sum = dapxsumkbn.main( 4, 5.0, x1, 2 );
// returns 25.0
```

> #### dapxsumkbn.ndarray( N, x, strideX, offsetX )

Reviewer: ```suggestion
#### dapxsumkbn.ndarray( N, alpha, x, strideX, offsetX )
```

Adds a scalar constant to each double-precision floating-point strided array element and computes the sum using an improved Kahan–Babuška algorithm and alternative indexing semantics.

```javascript
var Float64Array = require( '@stdlib/array/float64' );

var x = new Float64Array( [ 1.0, -2.0, 2.0 ] );

var sum = dapxsumkbn.ndarray( 3, 5.0, x, 1, 0 );
// returns 16.0
```

The function has the following additional parameters:

-   **offsetX**: starting index for `x`.

> While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to access every other value in `x` starting from the second value

Reviewer: ```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to access every other element starting from the second element:
```
